transcendence-app/src/chat/chatroom/infrastructure/db/postgres/chatroom.postgres.repository.ts

```diff
@@ -81,18 +83,43 @@ export class ChatroomPostgresRepository
       sort === BooleanString.True ? ChatroomKeys.NAME : ChatroomKeys.ID;
     const chatroomsData = await makeQuery<Chatroom>(this.pool, {
       text: `SELECT *
-      FROM ${this.table}
-      WHERE ${ChatroomKeys.NAME} ILIKE $1
-      ORDER BY ${orderBy}
-      LIMIT $2
-      OFFSET $3;`,
+             FROM ${this.table}
+             WHERE ${ChatroomKeys.NAME} ILIKE $1
+             ORDER BY ${orderBy}
+             LIMIT $2 OFFSET $3;`,
       values: [`%${search}%`, limit, offset],
     });
     return chatroomsData
       ? chatroomsData.map((chatroom) => new this.ctor(chatroom))
       : null;
   }
 
+  async getAuthUserPaginatedChatrooms(
+    authUserId: string,
+    queryDto: Required<PaginationWithSearchQueryDto>,
+  ): Promise<Chatroom[] | null> {
+    const { limit, offset, sort, search } = queryDto;
+    const orderBy =
+      sort === BooleanString.True
+        ? ChatroomMemberKeys.JOINED_AT
+        : ChatroomKeys.NAME;
+    const chatroomsData = await makeQuery<Chatroom>(this.pool, {
+      text: `SELECT *
+             FROM ${this.table} c
+                    INNER JOIN ${table.CHATROOM_MEMBERS} cm
+                               ON cm.${ChatroomMemberKeys.USERID} = $4
+                                 AND cm.${ChatroomMemberKeys.CHATID} = c.${ChatroomKeys.ID}
+                                 AND cm.${ChatroomMemberKeys.JOINED_AT} IS NOT NULL
```

Comment:
thought: For me, it's more readable if we move this condition to the where clause.

Reply:
done